server/services/survey.service.js

```diff
@@ -0,0 +1,60 @@
+const Path = require('path-parser');
+const { URL } = require('url');
+
+module.exports = class SurveyService {
+
+  async updateSurveys(data) {
+    try {
+
+      const uniqueServeys =
+          data.map(this.buildSurveyDataFromBody)
+              .filter(this.filterOutNullAndUndefined)
+              .filter(this.filterByUniqueEmailAndSurveyId);
+
+      for (let survey of uniqueServeys) {
+        await Survey.updateOne(
+            {
+              _id: survey.surveyId,
+              recipients: {
+                $elemMatch: {email: survey.email, responded: false}
+              }
+            },
+            {
+              $inc: {[survey.choice]: 1},
+              $set: {'recipients.$.responded': true},
+              lastResponded: new Date()
+            }
+        ).exec();
+      }
+
+      return { success: true }
+    } catch(error) {
+      return error;
+    }
+  }
+
+  getRecipientListFromEmailString(recipients) {
+    return recipients.split(',').map(email => {
+      return { email };
+    })
+  }
+
+
+  buildSurveyDataFromBody(bodyItem) {
+    const path = new Path('/api/surveys/:surveyId/:choice');
+    const match = path.test(new URL(surveyData.url).pathname);
+    if (match) {
+      return { email: surveyData.email, surveyId: match.surveyId, choice: match.choice };
+    }
+  }
+
+  filterOutNullAndUndefined(item) {
+    return item !== null && item !== undefined;
+  }
```

Comment:
That previous map function passes that array of data to this filter callback, which removes any null or undefined items. This should be impossible since the .map function above only adds items that match the URL, but since it was in the original code and I didn't actually test to be sure...I left it in. This can probably be removed though.